tests/sqlparser_postgres.rs

```diff
@@ -2509,6 +2509,340 @@ fn parse_create_anonymous_index() {
+        }) => {
+            assert_eq_vec(&["users_name_trgm_idx"], &name);
+            assert_eq_vec(&["users"], &table_name);
+            assert_eq!(IndexType::GIN, using);
+            assert_eq!(
+                IndexColumn {
+                    column: OrderByExpr {
+                        expr: Expr::Function(Function {
+                            name: ObjectName(vec![ObjectNamePart::Identifier(Ident {
+                                value: "concat_users_name".to_owned(),
+                                quote_style: None,
+                                span: Span::empty()
+                            })]),
+                            uses_odbc_syntax: false,
+                            parameters: FunctionArguments::None,
+                            args: FunctionArguments::List(FunctionArgumentList {
+                                duplicate_treatment: None,
+                                args: vec![
+                                    FunctionArg::Unnamed(FunctionArgExpr::Expr(Expr::Identifier(
+                                        Ident {
+                                            value: "first_name".to_owned(),
+                                            quote_style: None,
+                                            span: Span::empty()
+                                        }
+                                    ))),
+                                    FunctionArg::Unnamed(FunctionArgExpr::Expr(Expr::Identifier(
+                                        Ident {
+                                            value: "last_name".to_owned(),
+                                            quote_style: None,
+                                            span: Span::empty()
+                                        }
+                                    )))
+                                ],
+                                clauses: vec![]
+                            }),
+                            filter: None,
+                            null_treatment: None,
+                            over: None,
+                            within_group: vec![]
+                        }),
+                        options: OrderByOptions {
+                            asc: None,
+                            nulls_first: None,
+                        },
+                        with_fill: None,
+                    },
+                    operator_class: Some(Ident::new("gin_trgm_ops")),
+                },
+                columns[0],
+            );
+            assert!(!unique);
+            assert!(!concurrently);
+            assert!(!if_not_exists);
+            assert!(include.is_empty());
+            assert!(with.is_empty());
+        }
+        _ => unreachable!(),
+    }
+}
+
+#[test]
+fn parse_create_projects_name_description_trgm_index() {
+    let sql = "CREATE INDEX projects_name_description_trgm_idx ON projects USING GIN (concat_projects_name_description(name, description) gin_trgm_ops)";
```

Comment:
this test scenario looks identical to `parse_create_users_name_trgm_index` in terms of coverage?
Im thinking generally for the tests we can group them into the same test function, it would be sufficient with the AST assertion on only one of the test scenarios then for the rest we can rely only on `verified_stmt()`, that would keep the tests code smaller and easier to track what part of the syntax is covered.

Then could we add scenarios for all the introduced index types (i.e. BRIN, BLOOM etc)? As well as the custom type which probably would live in common I imagine?